rust/schema.py

```diff
@@ -31,30 +33,514 @@ class Location(Element):
     name: string
+    body: Expr
+
+
+#     Missing,
+class MissingExpr(Expr):
+    pass
+
+#     Path(Path),
+
+
+class Path(Expr):
+    # TODO
+    pass
+
+#     If {
+#         condition: ExprId,
+#         then_branch: ExprId,
+#         else_branch: Option<ExprId>,
+#     },
+
+
+class If(Expr):
+    condition: Expr
+    then_branch: Expr
+    else_branch: optional[Expr]
+
+#     Let {
+#         pat: PatId,
+#         expr: ExprId,
+#     },
+
+
+class Let(Expr):
+    pat: Pat
+    expr: Expr
+
+#     Block {
+#         id: Option<BlockId>,
+#         statements: Box<[Stmt]>,
+#         tail: Option<ExprId>,
+#         label: Option<LabelId>,
+#     },
+
+
+class Block(Expr):
+    statements: list[Stmt]
+    tail: optional[Expr]
+    label: optional[string]
+
+#     Async {
+#         id: Option<BlockId>,
+#         statements: Box<[Stmt]>,
+#         tail: Option<ExprId>,
+#     },
+
+
+class Async(Block):
+    pass
+
+#     Const(ConstBlockId),
+
+
+class Const(Expr):
+    pass
+
+#     // FIXME: Fold this into Block with an unsafe flag?
+#     Unsafe {
+#         id: Option<BlockId>,
+#         statements: Box<[Stmt]>,
+#         tail: Option<ExprId>,
+#     },
+
+
+class Unsafe(Block):
+    pass
+
+#     Loop {
+#         body: ExprId,
+#         label: Option<LabelId>,
+#     },
+
+
+class Loop(Expr):
+    body: Expr
+    label: optional[string]
+
+#     Call {
+#         callee: ExprId,
+#         args: Box<[ExprId]>,
+#         is_assignee_expr: bool,
+#     },
+
+
+class Call(Expr):
+    callee: Expr
+    args: list[Expr]
+    is_assignee_expr: predicate
+
+#     MethodCall {
+#         receiver: ExprId,
+#         method_name: Name,
+#         args: Box<[ExprId]>,
+#         generic_args: Option<Box<GenericArgs>>,
+#     },
+
+
+class MethodCall(Expr):
+    receiver: Expr
+    method_name: string
+    args: list[Expr]
+    # TODO
+    # generic_args: optional[GenericArgs]
+
+# pub struct MatchArm {
+#     pub pat: PatId,
+#     pub guard: Option<ExprId>,
+#     pub expr: ExprId,
+# }
+
+
+@qltest.skip
+class MatchArm(Locatable):
+    pat: Pat
+    guard: optional[Expr]
+    expr: Expr
+#     Match {
+#         expr: ExprId,
+#         arms: Box<[MatchArm]>,
+#     },
+
+
+class Match(Expr):
+    expr: Expr
+    branches: list[MatchArm]
+
+#     Continue {
+#         label: Option<LabelId>,
+#     },
+
+
+class Continue(Expr):
+    label: optional[string]
+
+#     Break {
+#         expr: Option<ExprId>,
+#         label: Option<LabelId>,
+#     },
+
+
+class Break(Expr):
+    expr: optional[Expr]
+    label: optional[string]
+
+
+#     Return {
+#         expr: Option<ExprId>,
+#     },
+
+class Return(Expr):
+    expr: optional[Expr]
+#     Become {
+#         expr: ExprId,
+#     },
+
+
+class Become(Expr):
+    expr: Expr
+#     Yield {
+#         expr: Option<ExprId>,
+#     },
+
+
+class Yield(Expr):
+    expr: optional[Expr]
+
+#     Yeet {
+#         expr: Option<ExprId>,
+#     },
+
+
+class Yeet(Expr):
```

Comment:
for the curious, this is an experimental feature that we're not probably going to really encounter https://doc.rust-lang.org/stable/unstable-book/language-features/yeet-expr.html

Reply:
Maybe we should just remove it for now? Given that it's highly experimental and placeholder syntax that is going to change in the future.

Reply:
Not sure. I could remove the class and translate `yeet` to `MissingExpr` in the extractor.